src/Components/Details/Details.tsx

```diff
@@ -1,69 +1,88 @@
-import './Details.css'
+import './Details.css';
 import Youtube from '../Youtube/Youtube';
 import Twitch from '../Twitch/Twitch';
 import { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { Creator , CreatorInfo, YoutubeVideo, TwitchVideo} from '../../types';
+import { Creator , CreatorInfo } from '../../types';
 import { getCreatorInfo } from '../../apiCalls';
+import { MdFavorite } from "react-icons/md";
+import { MdFavoriteBorder } from "react-icons/md";
 
 type DetailsProps = {
-	displayedCreators: Creator[] | null
-}
+	savedCreators: Creator[] | null,
+	favoriteCreator: (creatorInfo: CreatorInfo) => void,
+	unfavoriteCreator: (id: number) => void
+};
 
-function Details({ displayedCreators }: DetailsProps) {
+function Details({ savedCreators, favoriteCreator, unfavoriteCreator }: DetailsProps) {
 	const { id } = useParams();
-	const [selectedButton, setSelectedButton] = useState<string>('youtube');
+	const [selectedContentButton, setSelectedContentButton] = useState<string>('youtube');
 	const [creatorInfo, setCreatorInfo] = useState<CreatorInfo | null>(null);
+	const [favorited, setFavorited] = useState<boolean>(false);
 
-	const handleButtonClick = (buttonId: string) => {
-		setSelectedButton(buttonId);
+	const checkIfFavorited = () => {
+		if (savedCreators?.find(creator => creator.id === parseInt(id || '', 10))) {
+			setFavorited(true);
+		} else {
+			setFavorited(false);
+		};
 	};
 
+	const toggleHeart = (creatorInfo: CreatorInfo) => {
+		if (favorited) {
+			unfavoriteCreator(parseInt(creatorInfo.id));
+			setFavorited(false);
+		} else {
+			favoriteCreator(creatorInfo);
+			setFavorited(true);
+		};
+	};
 
-	let creator;
-	if (displayedCreators) {
-		creator = displayedCreators.find(creator => creator.id === parseInt(id || '', 10));
-	}
+	const handleContentButtonClick = (buttonId: string) => {
+		setSelectedContentButton(buttonId);
+	};
 
 	useEffect(() => {
-		const fetchData = async () => {
-			const data = await getCreatorInfo(parseInt(id || '', 10));
-			if (data) {
-				setCreatorInfo(data.data);
-			}
-		};
-	
-		fetchData();
+		getCreatorInfo(parseInt(id || '', 10))
+		.then(data => {
+			setCreatorInfo(data.data);
+			checkIfFavorited();
+		})
 	}, [id]);
 
 
 	return (
-		<div className='details'>
-			{creator && (
-				<>
-					<h2>{creator.name}'s Details</h2>
+		<div className='details-page'>
```

Comment:
Nice build on details-page with getting the toggleHeart functionality working. Utilizing icons as an import to the project gives this a much more refined style!